src/lib.rs

```diff
@@ -73,33 +69,176 @@ pub unsafe extern "C" fn pthread_join(
     native: libc::pthread_t,
     _value: *mut *mut libc::c_void,
 ) -> libc::c_int {
-    ctru_sys::threadJoin(native as ctru_sys::Thread, u64::MAX);
+    if native == MAIN_THREAD_ID {
+        // This is not a valid thread to join on
+        return libc::EINVAL;
+    }
+
+    let result = ctru_sys::threadJoin(native as ctru_sys::Thread, u64::MAX);
+    if ctru_sys::R_FAILED(result) {
+        // TODO: improve the error code by checking the result further?
+        return libc::EINVAL;
+    }
+
     ctru_sys::threadFree(native as ctru_sys::Thread);
 
     0
 }
 
 #[no_mangle]
 pub unsafe extern "C" fn pthread_detach(thread: libc::pthread_t) -> libc::c_int {
+    if thread == MAIN_THREAD_ID {
+        // This is not a valid thread to detach
+        return libc::EINVAL;
+    }
+
     ctru_sys::threadDetach(thread as ctru_sys::Thread);
 
     0
 }
 
 #[no_mangle]
-pub unsafe extern "C" fn pthread_getpriority() -> libc::c_int {
+pub unsafe extern "C" fn pthread_self() -> libc::pthread_t {
+    ctru_sys::threadGetCurrent() as libc::pthread_t
+}
+
+unsafe fn get_main_thread_handle() -> Result<ctru_sys::Handle, ctru_sys::Result> {
+    // Unfortunately I don't know of any better way to get the main thread's
+    // handle, other than via svcGetThreadList and svcOpenThread.
+    // Experimentally, the main thread ID is always the first in the list.
```

Comment:
Does `threadGetHandle(NULL)` not work?

Reply:
Wait, by “main thread”, do you mean Rust’s main or the `_start` function from libctru? I am pretty sure libctru spawns the application main on a different thread, and does the setup on the “actual” main. Test if `std::thread::current` is the same as this one.

Reply:
I found something interesting which maybe works instead of all this? 
https://libctru.devkitpro.org/svc_8h.html#a848b6094dcf31bfb0fa5cc999348c2d4